<?xml version="1.0"?>
<!--
デバッグ用の出力あり

===== Param =====
以下のように設定すること
- localizer
    flag_broadcast: true
    is_visible: true
- local_path_planner
    is_visible: true
-->
<launch>
    <!-- config file path -->
    <arg name="rviz_settings" default="$(find chibi23_b)/config/rviz/chibi23_b_test1.rviz"/>
    <arg name="map_settings" default="$(find chibi23_b)/map/map.yaml"/>
    <arg name="localizer_settings" default="$(find chibi23_b)/config/param/localizer.yaml"/>
    <!-- <arg name="obstacle_expander_settings" default="$(find chibi23_b)/config/param/obstacle_expander.yaml"/> -->
    <!-- <arg name="global_path_planner_settings" default="$(find chibi23_b)/config/param/global_path_planner.yaml"/> -->
    <!-- <arg name="local_goal_creator_settings" default="$(find chibi23_b)/config/param/local_goal_creator.yaml"/> -->
    <!-- <arg name="obstacle_detector_settings" default="$(find chibi23_b)/config/param/obstacle_detector.yaml"/> -->
    <!-- <arg name="local_map_creator_settings" default="$(find chibi23_b)/config/param/local_map_creator.yaml"/> -->
    <!-- <arg name="local_path_planner_settings" default="$(find chibi23_b)/config/param/local_path_planner.yaml"/> -->

    <!-- static tf -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_transform_publisher" args="0 0 0 0 0 0 1 /base_link /laser"/>

    <!-- Roomba -->
    <node pkg="roomba_500driver_meiji" type="main500" name="roomba_500driver_meiji"/>

    <!-- LiDAR -->
    <node pkg="urg_node" type="urg_node" name="urg_node" args="_serial_port:=/dev/ttyACM0"/>
    <node pkg="urg_node" type="urg_node" name="urg_node" args="_ip_address:=192.168.1.222"/>

    <!-- node launch -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_settings)"/>
    <node pkg="chibi23_b" type="localizer" name="localizer" output="screen">
        <rosparam command="load" file="$(arg localizer_settings)"/>
    </node>
    <node pkg="chibi23_b" type="local_goal_creator" name="local_goal_creator" output="screen">
        <!-- <rosparam command="load" file="$(arg local_goal_creator_settings)"/> -->
    </node>
    <!-- <node pkg="chibi23_b" type="obstacle_expander" name="obstacle_expander" output="screen"> -->
        <!-- <rosparam command="load" file="$(arg obstacle_expander_settings)"/> -->
    <!-- </node> -->
    <node pkg="chibi23_b" type="global_path_planner" name="global_path_planner" output="screen">
        <!-- <rosparam command="load" file="$(arg global_path_planner_settings)"/> -->
    </node>
    <node pkg="chibi23_b" type="obstacle_detector" name="obstacle_detector" output="screen">
        <!-- <rosparam command="load" file="$(arg obstacle_detector_settings)"/> -->
    </node>
    <node pkg="chibi23_b" type="local_map_creator" name="local_map_creator" output="screen">
        <!-- <rosparam command="load" file="$(arg local_map_creator_settings)"/> -->
    </node>
    <node pkg="chibi23_b" type="local_path_planner" name="local_path_planner" output="screen">
        <!-- <rosparam command="load" file="$(arg local_path_planner_settings)"/> -->
    </node>
    <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_settings)" output="screen"/>
</launch>
